BlueDBOnDisk/src/main/java/org/bluedb/disk/segment/SegmentSizeSettings.java

```diff
@@ -0,0 +1,99 @@
+package org.bluedb.disk.segment;
+
+import static java.util.Arrays.asList;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.bluedb.api.exceptions.BlueDbException;
+import org.bluedb.api.keys.BlueKey;
+import org.bluedb.api.keys.HashGroupedKey;
+import org.bluedb.api.keys.IntegerKey;
+import org.bluedb.api.keys.LongKey;
+import org.bluedb.api.keys.TimeKey;
+
+public enum SegmentSizeSettings {

```

Comment:
I think we need to decouple our public segment size options (`SegmentSize.java`) from our private options (`SegmentSizeSettings.java`). I think that we should be persisting the private `SegmentSizeSettings` configuration in the meta data of the collection instead of just a long so that we can be sure that we give the path manager the entire complete configuration that was originally used on the collection. This gives us flexibility to have more than one private configuration option for the same public option. We could change the default configuration that is used for a particular public option without breaking pre-existing collections that are still using an old configuration. We also won't have to update the public API in order to make a change.

Reply:
This is tricky.  I think we should actually commit to disk the size of the segment because we're actually already committing to that size in the folder structure.  Perhaps we shouldn't expose it in the public API though... we should discuss some time